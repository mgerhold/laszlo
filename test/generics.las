function join(array: [?], separator: String) ~> String {
    let result = "";
    if array.size > 0 {
        result = result + array[0];
    }
    for i in 1..array.size {
        result = result + separator + array[i];
    }
    return result;
}

println(join([1, 2, 3], ", "));
println(join(["test", "sentence", "with", "words"], ":"));
println(join([[1, 2, 3], [4, 5, 6], [7, 8, 9]], " <-> "));

function add(lhs: ?, rhs: ?) ~> ? {
    return lhs + rhs;
}

let a = add(1, 2);
let b = add("Hello, ", "World!");
let c = add("The answer is ", 42);

println(a);
assert(typeof(a) == "I32");
println(b);
assert(typeof(b) == "String");
println(c);
assert(typeof(c) == "String");
assert(typeof(add) == "Function(?, ?) ~> ?");

function f(n: I32) {
    let a = n;
    a = a + 1;
}

let words = ["a", "b"];
f(words.size);
