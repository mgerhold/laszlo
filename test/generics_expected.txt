1, 2, 3
test:sentence:with:words
[1, 2, 3] <-> [4, 5, 6] <-> [7, 8, 9]
3
Hello, World!
The answer is 42